### ISSUE TITLE
Failed to execute script 'ccdl' due to unhandled exception!

### ISSUE THREAD

**Issue by IanBauters** [chriswayg/CC-Offline-Package-Generator#12]
I keep getting the following error no matter what I try:

Parsing products.xml
Traceback (most recent call last):
  File "ccdl.py", line 655, in <module>
  File "ccdl.py", line 362, in runccdl
  File "ccdl.py", line 276, in parse_products_xml
TypeError: 'NoneType' object is not iterable
[2113] Failed to execute script 'ccdl' due to unhandled exception!

**Comment by buemaar** [comment]
Having the same issue here

**Comment by apparition47** [comment]
The base [adobe-packager](https://github.com/Drovosek01/adobe-packager) script has been updated to [0.1.4-hotfix1](https://github.com/Drovosek01/adobe-packager/blob/master/ccdl.py#L7) to fix this (similar to #11). You'll have to manually update the `/src` git submodule then follow [the build instructions](https://github.com/chriswayg/CC-Offline-Package-Generator#build-instructions) if you want to update, rebuild and use this app.

**Comment by smt311** [comment]
hard to follow the rebuild instructions.... please help to update the built version

**Comment by jonskoglund** [comment]
Still getting errors with [0.1.4-hotfix1](https://github.com/Drovosek01/adobe-packager/blob/master/ccdl.py#L7) when running the dmg.
```
/Volumes/CC_Offline_Package_Generator/CC_Offline_Package_Generator.app/Contents/Resources/CC-Offline-Package-Generator
=================================
= Adobe macOS Package Generator =
============= 0.1.4 =============

Do you want to make M1 native packages (y/N): y
Downloading products.xml

https://prod-rel-ffc-ccm.oobesaas.adobe.com/adobe-ffc-external/core/v4/products/all?_type=xml&channel=ccm,sti&platform=macuniversal,osx10-64,osx10&productType=Desktop
Source URL is: https://prod-rel-ffc-ccm.oobesaas.adobe.com/adobe-ffc-external/core/v4/products/all?_type=xml&channel=ccm,sti&platform=macuniversal,osx10-64,osx10&productType=Desktop
Traceback (most recent call last):
  File "urllib3/connection.py", line 174, in _new_conn
  File "urllib3/util/connection.py", line 95, in create_connection
  File "urllib3/util/connection.py", line 85, in create_connection
ConnectionRefusedError: [Errno 61] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "urllib3/connectionpool.py", line 703, in urlopen
  File "urllib3/connectionpool.py", line 386, in _make_request
  File "urllib3/connectionpool.py", line 1040, in _validate_conn
  File "urllib3/connection.py", line 358, in connect
  File "urllib3/connection.py", line 186, in _new_conn
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x7fa5a0636850>: Failed to establish a new connection: [Errno 61] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "requests/adapters.py", line 440, in send
  File "urllib3/connectionpool.py", line 785, in urlopen
  File "urllib3/util/retry.py", line 592, in increment
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='prod-rel-ffc-ccm.oobesaas.adobe.com', port=443): Max retries exceeded with url: /adobe-ffc-external/core/v4/products/all?_type=xml&channel=ccm,sti&platform=macuniversal,osx10-64,osx10&productType=Desktop (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fa5a0636850>: Failed to establish a new connection: [Errno 61] Connection refused'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "ccdl.py", line 665, in <module>
  File "ccdl.py", line 364, in runccdl
  File "ccdl.py", line 269, in get_products_xml
  File "ccdl.py", line 261, in r
  File "requests/sessions.py", line 542, in get
  File "requests/sessions.py", line 529, in request
  File "requests/sessions.py", line 645, in send
  File "requests/adapters.py", line 519, in send
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='prod-rel-ffc-ccm.oobesaas.adobe.com', port=443): Max retries exceeded with url: /adobe-ffc-external/core/v4/products/all?_type=xml&channel=ccm,sti&platform=macuniversal,osx10-64,osx10&productType=Desktop (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fa5a0636850>: Failed to establish a new connection: [Errno 61] Connection refused'))
[6609] Failed to execute script 'ccdl' due to unhandled exception!
```

**Comment by untitledpng** [comment]
+1 experiencing the same problem with the current build on a M1 Pro.

**Comment by dr3apap** [comment]
I fixed the error. I've make  a pull request so he can merge my patch. The error is TypeError exception, "dependencies" is a list object that is trying to append the result of finding a platform from the parsed xml but find return None if he find no match and None is not iterable since None is not an array of in a pythonic way "None" is not a list. So python throw exception and it propagate since the exception was not handle. It's just a case of wrapping that block in a try except or a quick hack to get it working is to wrap it in an if statement. Let me know if this help and I will be waiting for my patch to be merge.

**Comment by prestocrest** [comment]
I keep getting a similar error with the current build. Not quite sure how to rebuild using [0.1.4-hotfix1](https://github.com/Drovosek01/adobe-packager/blob/master/ccdl.py#L7). A merge of the patch would be much appreciated!

Error reads:

Parsing products.xml
Traceback (most recent call last):
  File "ccdl.py", line 655, in <module>
  File "ccdl.py", line 362, in runccdl
  File "ccdl.py", line 276, in parse_products_xml
TypeError: 'NoneType' object is not iterable
[44565] Failed to execute script 'ccdl' due to unhandled exception!

**Comment by tokxinyi** [comment]
Before the update happens, here is what you can try:

1. Click on [0.1.4-hotfix1](https://github.com/Drovosek01/adobe-packager/blob/master/ccdl.py#L7) and save this ccdl.py file.

2. Make sure you have xcode, python 3, git and homebrew installed in your system before rebuilding.

3. Run the following commands in Terminal to setup the environment.
```
cd ~
git clone --recurse-submodules https://github.com/chriswayg/CC-Offline-Package-Generator.git
cd ~/CC-Offline-Package-Generator
sudo true && ./build_prerequisites.sh
```

4. Open Finder > Go > Home > CC-Offline-Package-Generator > src
![How to access home folder in Mac](https://user-images.githubusercontent.com/31503073/187933043-366834f2-746d-4c0f-a88a-115ccc731e4e.png)

5. Replace ccdl.py file with the hotfix version.

6. Run the following commands in Terminal to rebuild CC_Offline_Package_Generator.dmg
```
cd ~/CC-Offline-Package-Generator
pipenv install
pipenv shell
pip list
./build_app.sh
```

**Comment by prestocrest** [comment]
I followed your instructions, but am not sure what should happen after step 6.

If I run the previously installed CC-Offline-Package-Generator at the end of the process, it results in a different but equally ccdl-related error:


=================================
= Adobe macOS Package Generator =
============= 0.1.4 =============

Do you want to make M1 native packages (y/N): y
Downloading products.xml

https://cdn-ffc.oobesaas.adobe.com/core/v4/products/all?_type=xml&channel=ccm,sti&platform=macuniversal,osx10-64,osx10&productType=Desktop
Source URL is: https://cdn-ffc.oobesaas.adobe.com/core/v4/products/all?_type=xml&channel=ccm,sti&platform=macuniversal,osx10-64,osx10&productType=Desktop
Source URL is: https://cdn-ffc.oobesaas.adobe.com/core/v4/products/all?_type=xml&channel=ccm,sti&platform=macarm64,macuniversal&productType=Desktop
Parsing products.xml
Traceback (most recent call last):
  File "ccdl.py", line 655, in <module>
  File "ccdl.py", line 362, in runccdl
  File "ccdl.py", line 276, in parse_products_xml
TypeError: 'NoneType' object is not iterable
[62154] Failed to execute script 'ccdl' due to unhandled exception!
***@***.*** ~ % 


To be frank, I don’t have much of a clue what I am doing there, what a virtual environment is etc. I don’t expect you to explain that to me, but if you happen to know what I’m doing wrong and how to do it better, I’d be glad to know.

**Comment by inspiredearth** [comment]
Whilst @chriswayg has done a nice job putting Adobe Packager into an installable DMG, it appears he's not been active on Github for ~2 years. 
![image](https://user-images.githubusercontent.com/667395/201452705-392b13c9-f336-41fd-acbb-a52bc99f1a2d.png)
Not very encouraging when it comes to having this issue in his git resolved.

It appears the easiest solution to this error is to simply use the current version of [Adobe Packager ](https://github.com/Drovosek01/adobe-packager)(0.1.4). It works fine, and doesn't have the `Failed to execute script 'ccdl' due to unhandled exception!` error occurring. 

The only issue I've encountered with it so far (and it might have been a one-off), is that it had an error when trying to "Download another package" after I'd downloaded an initial package. The solution was to start the script again.

**Comment by joster1969** [comment]
Also having this issue. Anyone solved it?

**Comment by Jordytgm10** [comment]
Run the install.sh file first.
